misk-hibernate/src/main/kotlin/misk/hibernate/TransformedColumnType.kt

```diff
@@ -0,0 +1,122 @@
+package misk.hibernate
+
+import org.hibernate.HibernateException
+import org.hibernate.engine.spi.SharedSessionContractImplementor
+import org.hibernate.type.spi.TypeConfiguration
+import org.hibernate.type.spi.TypeConfigurationAware
+import org.hibernate.usertype.ParameterizedType
+import org.hibernate.usertype.UserType
+import java.io.Serializable
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.sql.Types
+import java.util.Properties
+import kotlin.reflect.KClass
+import kotlin.reflect.full.primaryConstructor
+
+internal class TransformedColumnType : UserType, ParameterizedType, TypeConfigurationAware {
+  companion object {
+    val TRANSFORMER_CLASS = "transformerClass"
```

Comment:
Maybe I haven't digested this enough yet, but do you have a convention in mind for versioning transformers? I.e., should transformer classes never be deleted or changed, and all changes yield a new class? Or should transformers be version aware to enable backwards compatible changes to transformer business logic?

Reply:
Versioning would happen on top of transformers; this makes it able to create something like a `JsonColumn` or `ProtobufColumn` external to misk-hibernate.